packages/KSelect/KSelect.vue

```diff
@@ -347,21 +385,30 @@ export default {
       return this.placeholderText
     }
   },
-  beforeMount () {
-    // items need keys
-    this.selectItems = this.items
-
-    for (let i = 0; i < this.selectItems.length; i++) {
-      this.selectItems[i].key = `${this.selectItems[i].label.replace(' ', '-')}-${i}`
-      if (this.selectItems[i].value === this.value || this.selectItems[i].selected) {
-        this.selectItems[i].selected = true
-        this.selectedItem = this.selectItems[i]
-        this.selectItems[i].key += '-selected'
-
-        if (this.appearance === 'select') {
-          this.filterStr = this.selectedItem.label
+  watch: {
+    items: {
+      handler () {
+        // items need keys
+        this.selectItems = this.items
+
+        for (let i = 0; i < this.selectItems.length; i++) {
+          this.selectItems[i].key = `${this.selectItems[i].label.replace(' ', '-')}-${i}`
```

Comment:
issue: There's a chance other invalid characters could be passed, so let's use a regular expression
```suggestion
          this.selectItems[i].key = `${this.selectItems[i].label.replace(' ', '-').replace(/[^a-z0-9-_]/gi, '')}-${i}`
```